library/ssl_srv.c

```diff
@@ -869,7 +869,7 @@ static int ssl_ciphersuite_match( mbedtls_ssl_context *ssl, int suite_id,
     if( mbedtls_ssl_ciphersuite_uses_psk( suite_info ) &&
         ssl->conf->f_psk == NULL &&
         ( ssl->conf->psk == NULL || ssl->conf->psk_identity == NULL ||
```

Comment:
Should we also accept `ssl->conf->psk_identity == NULL` if `ssl->conf->psk_identity_len == 0`? Its not a major problem, but I am not sure why this is not the case...

Reply:
That's a good point. Should we always require a pointer to nothing? I don't think we should.

So that expression should be as follows, shouldn't it?
```
if( mbedtls_ssl_ciphersuite_uses_psk( suite_info ) &&
         ssl->conf->f_psk == NULL &&
         ( ssl->conf->psk == NULL && ssl->conf->psk_len > 0 ) ||
         ( ssl->conf->psk_identity == NULL && ssl->conf->psk_identity_len > 0 ) )
```

Reply:
@sbutcher-arm I'm not sure what should be the behaviour here - if we should skip or fail on an inconsistency, and I didn't investigate where psk_identity exactly comes from, but I'd like to propose a different set of checks:
```
if( mbedtls_ssl_ciphersuite_uses_psk( suite_info ) &&
	ssl->conf->f_psk == NULL &&
	( ssl->conf->psk == NULL || ssl->conf->psk_len == 0 ||
	  ( ssl->conf->psk_identity == NULL && ssl->conf->psk_identity_len > 0 ) ||
	  ( ssl->conf->psk_identity != NULL && ssl->conf->psk_identity_len == 0 )
	 ) )
{
```
This way we ensure that either both psk_identity is non null and psk_identity_len is greater than zero, or that psk_identity is null and psk_identity_len equals zero, hence giving us a consistent state. 
I don't think that we should enable passing a valid pointer and zero length, or passing null with length, since this may be the users unnoticed mistake.

Reply:
@AndrzejKurek - I don't think we have to be that strict. We're following the convention that if `psk_len == 0` then `psk` *may* be `NULL`. It doesn't have to be.